### DISCUSSION TITLE
EMQX packagecloud.io repo for Debian 12 Bookworm

### DISCUSSION THREAD

**Discussion by alexdelprete** [emqx/emqx#11176] [Q&A] [answered]
Hi,

I upgraded my EMQX server to Debian 12. I also updated the `/etc/apt/sources.list.d/emqx_emqx.list`:

```
deb [signed-by=/etc/apt/keyrings/emqx_emqx-archive-keyring.gpg] https://packagecloud.io/emqx/emqx/debian/ bookworm main
deb-src [signed-by=/etc/apt/keyrings/emqx_emqx-archive-keyring.gpg] https://packagecloud.io/emqx/emqx/debian/ bookworm main
```

When I `apt-get update` I have the following error:

```
Hit:1 http://security.debian.org/debian-security bookworm-security InRelease
Hit:2 http://ftp.debian.org/debian bookworm InRelease
Get:3 http://ftp.debian.org/debian bookworm-updates InRelease [52,1 kB]
Hit:4 http://ftp.ch.debian.org/debian bookworm InRelease
Hit:5 http://ftp.it.debian.org/debian bookworm InRelease
Hit:6 https://ftp.linux.it/debian bookworm InRelease                                                                                                            
Hit:7 https://debian.ethz.ch/debian bookworm InRelease                                                                                    
Hit:8 https://alpha.floxsdlc.com/downloads/debian-archive unstable/ InRelease                                       
Hit:9 https://deb.volian.org/volian scar InRelease                    
Ign:10 https://packagecloud.io/emqx/emqx/debian bookworm InRelease
Err:11 https://packagecloud.io/emqx/emqx/debian bookworm Release
  404  Not Found [IP: 54.67.35.197 443]
Reading package lists... Done
E: The repository 'https://packagecloud.io/emqx/emqx/debian bookworm Release' does not have a Release file.
N: Updating from such a repository can't be done securely, and is therefore disabled by default.
```

I'll probably leave the source file pointing to bullseye while waiting for this to be solved, if there's another workaround please let me know.

Thanks.

**Comment by id** [comment]
@alexdelprete thanks for the report! We are not building Debian 12 packages as of now. I'll follow on this soon.

**Reply by alexdelprete** [reply]
Thank you for the help.

**Comment by alexdelprete** [comment]
@id Ivan, can I try using the Debian 11 .deb file on Debian 12 or will it mess up things? I wanted to upgrade to v5.1.1.

**Comment by id** [comment]
@alexdelprete we haven't tested this, so I can't really guarantee anything.
Wouldn't recommend this for production if you need to ensure reliability.

**Reply by alexdelprete** [reply]
Ok, I'll wait for Debian 12 packages then. Thank you.

**Comment by id** [comment, marked as answer]
Debian 12 packages are available since https://github.com/emqx/emqx/releases/tag/v5.1.2.

**Reply by alexdelprete** [reply]
> Debian 12 packages are available since https://github.com/emqx/emqx/releases/tag/v5.1.2.

Sorry to reopen this: I tried to upgrade to v5.3.1 on Debian 12, it only finds EMQX `v5.3.1-alpha.1`

I checked on packagecloud.io repo and it seems everything ok, but my system can't find versions after `alpha.1`.

This is my emqx source list:

```
# this file was generated by packagecloud.io for
# the repository at https://packagecloud.io/emqx/emqx

deb [signed-by=/etc/apt/keyrings/emqx_emqx-archive-keyring.gpg] https://packagecloud.io/emqx/emqx/debian/ bookworm main
deb-src [signed-by=/etc/apt/keyrings/emqx_emqx-archive-keyring.gpg] https://packagecloud.io/emqx/emqx/debian/ bookworm main
```

**Reply by id** [reply]
@alexdelprete it still says "Indexing" on packagecloud.io, give it a few hours and try again?

**Reply by alexdelprete** [reply]
@id before upgrading I checked the list and I saw the package available on packagecloud, so I assumed everything was ok. Wrong assumption.

What I don't like is that unstable releases are installed when upgrading, now I downloaded the deb package and installed it manually.

Is there a way to avoid apt-get installing unstable releases when upgrading?

**Reply by id** [reply]
It's on us, we should stop publishing pre-release packages to the same repo. I plan to fix this soon.

**Reply by alexdelprete** [reply]
I understand. Thanks for the feedbacks.
